From 19e5b427abb9454d7b3afe368c049d6b70c0131f Mon Sep 17 00:00:00 2001
From: Andra-Teodora Ilie <andra.ilie@nxp.com>
Date: Wed, 8 Jun 2022 15:09:32 +0300
Subject: [PATCH 28/49] s32: dt-bindings: Define A53 minimum frequency

Lower frequency values are not supported by
a53_core_div_clk and a53_core_div10_clk clocks.

Issue: ALB-8876
Upstream-Status: Pending 

Signed-off-by: Andra-Teodora Ilie <andra.ilie@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 include/dt-bindings/clock/s32gen1-clock-freq.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/dt-bindings/clock/s32gen1-clock-freq.h b/include/dt-bindings/clock/s32gen1-clock-freq.h
index e1b013b74..23115dec3 100644
--- a/include/dt-bindings/clock/s32gen1-clock-freq.h
+++ b/include/dt-bindings/clock/s32gen1-clock-freq.h
@@ -22,6 +22,7 @@
 
 #if defined(PLAT_s32g2)
 #define S32GEN1_A53_MAX_FREQ			(1000 * MHZ)
+#define S32GEN1_A53_MIN_FREQ			(48 * MHZ)
 #define S32GEN1_ARM_PLL_VCO_MAX_FREQ		(2000 * MHZ)
 #define S32GEN1_ARM_PLL_PHI0_MAX_FREQ		(1000 * MHZ)
 #define S32GEN1_A53_FREQ			(1000 * MHZ)
@@ -33,6 +34,7 @@
 #define S32GEN1_QSPI_CLK_FREQ			(200 * MHZ)
 #elif defined(PLAT_s32g3)
 #define S32GEN1_A53_MAX_FREQ			(1300 * MHZ)
+#define S32GEN1_A53_MIN_FREQ			(48 * MHZ)
 #define S32GEN1_ARM_PLL_VCO_MAX_FREQ		(2600 * MHZ)
 #define S32GEN1_ARM_PLL_PHI0_MAX_FREQ		(1300 * MHZ)
 #define S32GEN1_A53_FREQ			(1300 * MHZ)
@@ -45,6 +47,7 @@
 
 #elif defined(PLAT_s32r)
 #define S32GEN1_A53_MAX_FREQ			(800 * MHZ)
+#define S32GEN1_A53_MIN_FREQ			    (38400000UL)
 #define S32GEN1_ARM_PLL_VCO_MAX_FREQ		(1600 * MHZ)
 #define S32GEN1_ARM_PLL_PHI0_MAX_FREQ		(800 * MHZ)
 #define S32GEN1_A53_FREQ			(800 * MHZ)
-- 
2.25.1

